src/field/fieldperp.cxx

```diff
@@ -143,8 +168,28 @@ FPERP_OP_REAL(-=, -);
 FPERP_OP_REAL(*=, *);
 FPERP_OP_REAL(/=, /);
 
+const IndexRange FieldPerp::region(REGION rgn) const {
+  switch (rgn) {
+  case RGN_ALL:
```

Comment:
It looks strange not having ``RGN_ALL`` implemented. If I understand right, it works because the switch drops into ``case RGN_NOZ`` which (at the moment) is the same as ``RGN_ALL``. If this changes at some point, we might catch a potential mistake by adding an explicit implementation of ``RGN_ALL`` now.

Reply:
Yes, I'm not these regions are as good as they could be. 

For example the y indices are taken to be {0,0}, which is fine for use with other perps but no good if we wanted to use this with Field3D indexing.

I would be happy to "fix" this in this PR if others think it would make sense. (I should note whilst the diff makes it look like I've added this as new code I haven't there's a corresponding "removal" further down, I just repositioned it in the file).